src/commonMain/kotlin/io/ktor/io/ByteArrayOperations.kt

```diff
@@ -0,0 +1,122 @@
+package io.ktor.io
+
+/**
+ * Loads [Short] from the byte array at the specified [index].
+ *
+ * @throws IndexOutOfBoundsException if there are not enough bytes in the array.
+ */
+public fun ByteArray.loadShortAt(index: Int): Short {
```

Comment:
why it's public?

Reply:
It looks like there are many libraries with similar functionality. I'm thinking if we should provide public extensions for the ByteArray. What do you think?

Reply:
looks legit
maybe then we should move those extensions in separate package, rather than root one?

Reply:
I would like to keep as few packages as possible because it works much better with auto-import. Could you tell me why should we move it to a separate package?

Reply:
If we provide users `Buffer` as some high-level API, such extensions on `ByteArray` in my mind considered as more low-level API and they will be not needed for majority of users, but still will be useful for experienced users. And as `ByteArray` still general API, I would not pollute high-level package with low-level details.

Reply:
I would propose to leave it as is for now, but create an issue to revisit it later, based on future work and it's usage or even external users feedback

Reply:
https://github.com/ktorio/ktor-io/issues/10